updater.go

```diff
@@ -57,11 +58,19 @@ func NewUpdater(source UpdateSource, config Config, log logging.Logger) *Updater
 }
 
 // Update checks, downloads and performs an update
-func (u *Updater) Update(ctx Context) (*Update, error) {
+func (u *Updater) Update(ctx Context) (*Update, *Error) {
```

Comment:
Yeah, sure, exposing custom error types in a public interface isn't common... and totally unnecessary here